cmd/storer2elastic/dataprocessor/dataReplayer_test.go

```diff
@@ -0,0 +1,307 @@
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewDataReplayer(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		argsFunc func() dataprocessor.DataReplayerArgs
+		exError  error
+	}{
+		{
+			name: "NilShardCoordinator",
+			argsFunc: func() dataprocessor.DataReplayerArgs {
+				args := getDataReplayArgs()
+				args.ShardCoordinator = nil
+				return args
+			},
+			exError: dataprocessor.ErrNilShardCoordinator,
+		},
+		{
+			name: "NilDatabaseReader",
+			argsFunc: func() dataprocessor.DataReplayerArgs {
+				args := getDataReplayArgs()
+				args.DatabaseReader = nil
+				return args
+			},
+			exError: dataprocessor.ErrNilDatabaseReader,
+		},
+		{
+			name: "NilMarshalizer",
+			argsFunc: func() dataprocessor.DataReplayerArgs {
+				args := getDataReplayArgs()
+				args.Marshalizer = nil
+				return args
+			},
+			exError: dataprocessor.ErrNilMarshalizer,
+		},
+		{
+			name: "NilHasher",
+			argsFunc: func() dataprocessor.DataReplayerArgs {
+				args := getDataReplayArgs()
+				args.Hasher = nil
+				return args
+			},
+			exError: dataprocessor.ErrNilHasher,
+		},
+		{
+			name: "NilUint64ByteSliceConverter",
+			argsFunc: func() dataprocessor.DataReplayerArgs {
+				args := getDataReplayArgs()
+				args.Uint64ByteSliceConverter = nil
+				return args
+			},
+			exError: dataprocessor.ErrNilUint64ByteSliceConverter,
+		},
+		{
+			name: "NilHeaderMarshalizer",
+			argsFunc: func() dataprocessor.DataReplayerArgs {
+				args := getDataReplayArgs()
+				args.HeaderMarshalizer = nil
+				return args
+			},
+			exError: dataprocessor.ErrNilHeaderMarshalizer,
+		},
+		{
+			name: "All arguments ok",
+			argsFunc: func() dataprocessor.DataReplayerArgs {
+				return getDataReplayArgs()
+			},
+			exError: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := dataprocessor.NewDataReplayer(tt.argsFunc())
+			require.Equal(t, err, tt.exError)
+		})
+	}
+}
+
+func TestDataReplayer_Range_NilHandlerShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := getDataReplayArgs()
+	dr, _ := dataprocessor.NewDataReplayer(args)
+
+	err := dr.Range(nil)
+	require.Equal(t, dataprocessor.ErrNilHandlerFunc, err)
+}
+
+func TestDataReplayer_Range_NoDatabaseFoundShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := getDataReplayArgs()
+	dr, _ := dataprocessor.NewDataReplayer(args)
+
+	handlerFunc := func(persistedData data.RoundPersistedData) bool {
+		return true
+	}
+	err := dr.Range(handlerFunc)
+	require.Equal(t, dataprocessor.ErrNoMetachainDatabase, err)
+}
+
+func TestDataReplayer_Range_EpochStartMetaBlockFoundForEpochStartKeyShouldWork(t *testing.T) {
+	t.Parallel()
+
+	marshalizer := &mock.MarshalizerMock{}
+
+	shard0HdrHash := []byte("sh0Hdr")
+	txMbHash := []byte("txMbHash")
+	rwdTxMbHash := []byte("txMbHash")
```

Comment:
just for consistency:
`rwdTxMbHash := []byte("rwdTxMbHash")`

Reply:
done